drivers/hmis/app/graphql/concerns/graphql_permission_checker.rb

```diff
@@ -22,7 +18,14 @@ def self.current_permission_for_context?(context, permission:, entity:)
         context.dataloader.with(Sources::ActiveRecordAssociation, association).load(record)
       end
     end
-    raise "Missing loader for #{entity.class.name}##{entity.id}" unless loader
+    raise "Missing loader for #{entity.class.name}##{entity.id}" unless loader && subject
+
+    data_source_id = if subject.is_a? GrdaWarehouse::DataSource
+      subject.id
+    else
+      subject.data_source_id
+    end
+    return false unless current_user.hmis_data_source_id == data_source_id
```

Comment:
Can you add some comments here about (1) `subject` is expected to be one of data source, project, organization, and (2) return false because even if the user has permission to view this record in another hmis data source, they should not be able to resolve it in the context of the hmis they are currently using (hmis_data_source_id)

This code is tricky so any comments will surely help us in the future :)